### PR TITLE
feat(plugin): add degradation switch for traffic control plugin

### PR DESCRIPTION
**Pull request by okaybase**
### What this PR does / why we need it:

Add degradation switch for traffic control plugin.
When the limit function is temporarily unavailable(e.g. redis timeout for limit-count plugin), we can auto allow requests to continue exec by enable degradation switch, this is important for some scenarios(hope the request continue exec where upstream services can handle it).

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [x] Have you modified the corresponding document?
* [x] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: apisix/plugins/limit-conn/init.lua [hunk 1]
```diff
@@ -41,7 +41,11 @@ function _M.increase(conf, ctx)
     local lim, err = lrucache(conf, nil, create_limit_obj, conf)
     if not lim then
         core.log.error("failed to instantiate a resty.limit.conn object: ", err)
-        return 500
+        if conf.degradation then
+            return
+        else
+            return 500
+        end
```
Comment (nic-chen) [inline apisix/plugins/limit-conn/init.lua:48]:
```suggestion
        if conf.degradation then
            return
        end
        return 500
```

**Review by nic-chen — APPROVED** [review]
LGTM

**Comment by okaybase** [comment]
> LGTM

Thanks for your review~ @nic-chen

**Review by tokers — APPROVED** [review]

File: apisix/plugins/limit-conn.lua [hunk 2]
```diff
@@ -33,6 +33,7 @@ local schema = {
         rejected_code = {
             type = "integer", minimum = 200, maximum = 599, default = 503
         },
+        degradation = {type = "boolean", default = false}
```
Comment (spacewander) [inline apisix/plugins/limit-conn.lua:36]: Would be better to use "allow_degradation"?

File: t/plugin/limit-count-redis-cluster.t [hunk 3]
```diff
@@ -377,3 +377,62 @@ code: 503
...
+passed
+--- no_error_log
+[error]
+
+
+
+=== TEST 11: enable degradation switch for TEST 10
+--- request
+GET /hello
+--- error_code eval
```
Comment (spacewander) [inline t/plugin/limit-count-redis-cluster.t:435]: We don't need to check 200 code - it is checked by default

**Review by spacewander — COMMENTED** [review]
Need a test in t/plugin/limit-conn.t

File: apisix/plugins/limit-conn.lua (hunk 2 above)
Comment (okaybase) [reply apisix/plugins/limit-conn.lua]: done

File: t/plugin/limit-count-redis-cluster.t (hunk 3 above)
Comment (okaybase) [reply t/plugin/limit-count-redis-cluster.t]: done

**Review by spacewander — COMMENTED** [review]
LGTM. Let's add new test in t/plugin/limit-conn.t

**Comment by okaybase** [comment]
> LGTM. Let's add new test in t/plugin/limit-conn.t

new test added, and improve a log info. @spacewander

**Review by spacewander — APPROVED** [review]

**Pull request merged**